FortranCon2021-stdlib/stdlib-talk.tex

```diff
@@ -0,0 +1,302 @@
+    Currently, on every pull request, the CI builds stdlib and runs its test suite on seven different platforms meaning different combinations of compiler, OS, and CPU architecture.
+    However, just because your platform is not shown here doesn't mean stdlib won't work on your system.
+    Anecdotally, I've built stdlib with several versions of ifort on a recent MacBook as well as several versions of gfortran on a 10-year-old netbook running 32-bit Linux.
+  }
+\end{frame}
+
+
+\begin{frame}
+  \frametitle{Room for improvement}
+  
+  \begin{itemize}
+  \item Fill out numerical capabilities
+    \begin{itemize}
+    \item ``Simple'' functions are often not so simple (e.g., $cbrt$)
+    \item Difficult to find reviewers with domain knowledge (see: Probability Distributions)
+    \item What to put in stdlib versus create fpm package?
+    \end{itemize}
+    
+  \item Improve consistency of documentation
+    \begin{itemize}
+    \item Lots of variability in style \& level of detail
+    \item To be addressed with standardized tempates
+    \end{itemize}
+  \end{itemize}
+
+  \note{
+    I've used this talk mainly to highlight the cool new developments and accomplishments of the stdlib project so far, but there is certainly room for improvement.
+    For one, most of the big exciting new modules in stdlib serve general programming needs.
+    There are a couple reasons that numerical functionality has been slower to come in.
+    One reason is that when one starts discussing the API and possible implementations, many ``simple'' functions turn out to have lots of subtle difficulties.
+    My first-hand experience with this was in a discussion of a possible cube-root function, where one has to consider all kinds of floating-point edge cases that one wouldn't bother with in personal code but which is important to get right in a ``standard'' library.
+    Another issue is that many numerical algorithms should be reviewed by people with particular domain knowledge to ensure that the implementation and tests make sense.
+    The prime example of this right now is a handful of pull requests for statistical distributions that have been waiting for a reviewer with an appropriate background in statistics.
+    When discussing new feature proposals, a recurring question is, ``should this be in stdlib or should this be an fpm package?''
+    Really, this is a testament to the progress made with fpm, but the point is that this boundary is still being mapped out.
+    The other area where the stdlib project is actively trying to improve things is in the documentation.
+    Even though documentation is required for every new procedure or type, the style and rigor of this documentation varies quite a lot.
+    The proposed solution right now is to establish a baseline template for specifications to follow.
+    This will hopefully give the stdlib documentation a more uniform feel, as well as be a helpful starting point for new contributors.
+  }
+\end{frame}
+
```

Comment:
Here's a draft roadmap slide, largely based on the current backlog of PRs, which I think is realistic. Please adapt it as you see fit.

```suggestion


\begin{frame}
  \frametitle{Outlook: Next 12 months}

  \begin{itemize}
  \item Probability distributions: Uniform, normal, exponential, gamma, and beta (Fisher)
  \item Generic linked list (Karwa)
  \item Generic map type (Ehlert)
  \item Hash functions (Clodius, Ehlert)
  \item Improved OS and file system facilities (Diehl, Markus)
  \item Selection algorithms
  \item All new intrinsics planned for Fortran 202X
  \item Improved stdlib test suite (Ehlert)
  \end{itemize}

  \note{Here are some additions and improvements to look forward to in the next year or so.
        The highlights are: procedures for various probability distributions and sampling
        pseudo-random numbers from them; derived types for generic collections
        (linked lists, see Chetan's talk today, and a map/dictionary type);
        and cross-platform procedures for querying the operating system and manipulating
        the file system (such as listing and traversing directories, removing or copying files, etc.).
        This roadmap is driven by community and their contributions and is largely based on
        the backlog of existing PRs to be reviewed and improved upon.
        If there's something you'd like to be a part of stdlib, please let us know by opening
        a proposal in GitHub issues, and we'll discuss it and come up with an implementation plan.
  }

\end{frame}


```

Reply:
Thank you @milancurcic for this roadmap. I would suggest to add to this list: "portability across platforms", or something similar.